src/app/delegation_backend/src/itn_uptime_analyzer/identity.go

```diff
@@ -0,0 +1,146 @@
+package itn_uptime_analyzer
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	dg "block_producers_uptime/delegation_backend"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	logging "github.com/ipfs/go-log/v2"
+	sheets "google.golang.org/api/sheets/v4"
+)
+
+// The graphQLPort parameter should be optional but Golang doesn't permit that
+// As a workaround we use a pointer that can be defined as nil
+type Identity struct {
+	id, publicKey, publicIp, uptime string
+	graphQLPort                     *string
+}
+
+// Custom function to check if identity is in array
+func IsIdentityInArray(id string, identities []Identity) bool {
+	for _, identity := range identities {
+		if identity.id == id {
+			return true
+		}
+	}
+	return false
+}
+
+// Goes through each submission and adds an identity type to a map
+// Identity is constructed based on the payload that the BP sends which may hold pubkey, ip address and graphqlport
+func CreateIdentities(config AppConfig, sheet *sheets.Service, ctx dg.AwsContext, log *logging.ZapEventLogger, sheetTitle string, currentTime time.Time, executionInterval int) []Identity {
+
+	currentDate := currentTime.Format("2006-01-02")
+	lastExecutionTime := GetLastExecutionTime(config, sheet, log, sheetTitle, currentTime, executionInterval)
+
+	prefixCurrent := strings.Join([]string{ctx.Prefix, "submissions", currentDate}, "/")
```

Comment:
Possibly a bug: new identities may have occurred in the previous date (within time range in question)